meteor/client/ui/SegmentTimeline/Parts/SegmentTimelinePart.tsx

```diff
@@ -306,30 +310,47 @@ export class SegmentTimelinePartClass extends React.Component<Translated<WithTim
 	}
 
 	getLayerStyle(): React.CSSProperties {
-		const actualPartDuration = SegmentTimelinePartClass.getPartDuration(
-			this.props,
-			this.state.liveDuration,
-			this.state.isDurationSettling,
-			this.state.durationSettlingStartsAt
+		const partDisplayDuration = SegmentTimelinePartClass.getPartDisplayDuration(
+			this.props.part,
+			this.props.timingDurations
 		)
 		const partDuration = this.props.cropDuration
-			? Math.min(this.props.cropDuration, actualPartDuration)
-			: actualPartDuration
+			? Math.min(this.props.cropDuration, partDisplayDuration)
+			: partDisplayDuration
+		const partDurationWithFutureShade = partDuration + (this.state.isLive ? this.getFutureShadePaddingTime() : 0)
 		if (this.props.relative) {
 			return {
-				width: ((partDuration / (this.props.totalSegmentDuration || 1)) * 100).toString() + '%',
+				width: ((partDurationWithFutureShade / (this.props.totalSegmentDuration || 1)) * 100).toString() + '%',
 			}
 		} else {
 			return {
-				minWidth: Math.round(partDuration * this.props.timeScale).toString() + 'px',
+				minWidth: (partDurationWithFutureShade * this.props.timeScale).toString() + 'px',
 			}
 		}
 	}
 
 	getPartStyle(): React.CSSProperties {
-		return this.getLayerStyle()
+		const style = this.getLayerStyle()
+
+		let timeOffset = SegmentTimelinePartClass.getPartStartsAt(this.props)
+		if (this.props.isLiveSegment && this.props.anyPriorPartWasLive && !this.state.isLive) {
+			timeOffset += this.getFutureShadePaddingTime()
+		}
+
+		return {
+			...style,
+			transform: `translateX(${this.convertTimeToPixels(timeOffset)}px)`,
+			willChange: this.props.isLiveSegment
+				? ['transform', this.props.relative ? 'width' : 'min-width'].join(', ')
+				: 'none',
+		}
+	}
+
+	private convertTimeToPixels = (time: number) => {
+		return this.props.timeScale * time
```

Comment:
The quickfix would be to rename `props.timeScale` to something like `props.timeToPixelRatio`.
Then this method would make a lot of sense:
```
private convertTimeToPixels = (time: number) => {
  return this.props.timeToPixelRatio * time
}
```

It would also make `getFutureShadePaddingPixels` make more sense